xocto/ranges.py

```diff
@@ -945,3 +946,56 @@ def any_overlapping(ranges: Iterable[Range[T]]) -> bool:
             return True
         range_set.add(range)
     return False
+
+
+def as_finite_datetime_periods(
+    periods: Iterable[HalfFiniteDatetimeRange | DatetimeRange],
```

Comment:
🧀 One source of multiple periods is where we're working with `RangeSet`s, which are essentially lists of periods.